src/database/mod.rs

```diff
@@ -310,6 +311,45 @@ impl Database {
             .into_iter()
             .collect())
     }
+
+    pub async fn get_batched_commitments(
+        &self,
+        last_processed_hash: &Hash,
+    ) -> Result<Vec<TreeUpdate>, Error> {
+        let mut tx = self.pool.begin().await?;
+
+        let batch_entry = tx.get_next_batch(last_processed_hash).await?;
+        let Some(batch_entry) = batch_entry else {
+            return Ok(vec![]);
+        };
+
+        let batches = tx.get_all_batches_after(batch_entry.id).await?;
+        let mut result = vec![];
+        for batch in batches {
+            match batch.batch_type {
```

Comment:
Can't we have the same code for both cases? In case for deletion we replace `*batch.commitments.0.get(i).unwrap()` with `Hash::ZERO` but `*batch.commitments.0.get(i).unwrap()` should be zero anyway, no?